packages/manager/modules/manager-pci-common/src/contexts/PCICommonContext/PCICommonContext.ts

```diff
@@ -0,0 +1,20 @@
+import { createContext } from 'react';
+
+export interface InternalMeta {
+  has3AZ?: boolean;
```

Comment:
any reason to use a context to store this data ?

if we use tanstack query you can directly fetch the data in the useHas3AZ hook thanks to the cache it will only be called once, it makes things easier to manage as you don't have to rely on a react context & factory

Reply:
the 3AZ condition depends on the product universe where it is called (product availability api call) so the context is provided from the application above and pass in a context to avoid prop drilling and anticipate future updates where this condition will be removed

Reply:
You can see an example on usage within block storage here : https://github.com/ovh/manager/pull/14406/files#diff-fea4f807a564699ba0090be979e28290cf2c0abac7d7db910c44f4020546d8a1R25-R53